Eatery/Controllers/Eateries/Campus/ScrollableViewController.swift

```diff
@@ -0,0 +1,374 @@
+//
+//  ScrollableViewController.swift
+//  Eatery
+//
+//  Created by Sergio Diaz on 12/9/20.
+//  Copyright © 2020 Cornell AppDev. All rights reserved.
+//
+
+import UIKit
+
+class ScrollableViewController: UIViewController {
+
+    private var eatery: CampusEatery!
+    private var viewControllers: [CampusEateryExpandedMenuViewController]!
+    private var allItems: [ExpandedMenu.Item]!
+
+    private var originalViews: [UIView]!
+    private var tabBar: TabBar!
+    private var tabBarStack: UIStackView!
+    private var menuItemsStack: UIStackView!
+    private var menuItemsHighToLowStack: UIStackView!
+    private var menuItemsLowToHighStack: UIStackView!
+
+    private var menuItemsView: UIView!
+    private var filterButton: ExtendedFilterButton!
+    private var filterLabel: UILabel!
+    private var headerContainer: UIView!
+    private var separatorView: SeparatorView!
+
+    let tabBarPadding: CGFloat = 16
+    var scrollView: UIScrollView?
+    var scrollOffset: CGFloat = 0
+    var fillerSpaceHeight: CGFloat = 0
+
+    private var beingMoved = false
+    private var isManualScrolling = false
+    private var orderedViewsCreated = false
+
+    init(eatery: CampusEatery, viewControllers: [CampusEateryExpandedMenuViewController], items: [ExpandedMenu.Item]) {
+        super.init(nibName: nil, bundle: nil)
+
+        self.eatery = eatery
+        self.viewControllers = viewControllers
+        self.originalViews = viewControllers.map {
+            $0.view
+        }
+
+        self.allItems = Sort.expandedMenuMergeSort(items)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .wash
+
+        createTabBarStack()
+        createMenuStack()
+        orderedViewsCreated = false
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        let screenSize = UIScreen.main.bounds.height
+        let tabBarHeight = (self.tabBarController?.tabBar.frame.height) ?? 0
+        let smallOffset: CGFloat = CampusEateryExpandedMenuViewController.heightConst
+
+        // Scroll Offset is used below because it accounts for NavBar height and edge insets
+        // reducedScreenSize accounts for the height of the screen minus all nav and tab bars
+        let reducedScreenSize = screenSize + (scrollOffset - tabBarHeight) + smallOffset
+        if originalViews.count > 0 {
+            let finalView = originalViews[originalViews.count - 1]
+            let checkHeight = reducedScreenSize - (menuItemsStack.frame.maxY - finalView.frame.minY)
+            fillerSpaceHeight = checkHeight > 0 ? checkHeight : 0
+        }
+
+        self.scrollView?.contentSize.height += fillerSpaceHeight
+    }
+
+    private func createTabBarStack() {
+        headerContainer = UIView()
+        headerContainer.layer.zPosition = 3
```

Comment:
What does this line do? Not too sure how layer `zPosition` works

Reply:
The zPosition property allows for the `HeaderContainer` to be on top of the `ScrollableTextTabBar` when the user decides to sort all items.  Otherwise, the `ScrollableTextTabBar` would be on top when the user sorts everything.

I believe I could've also fixed this by adding the `HeaderContainer` to its superview after adding the `ScrollableTextTabBar`.  However, I decided to do it like this because it would make more sense to add these view in the order they appear on the phone screen.  I could change this though if you'd like.

Reply:
to avoid confusion with what goes on top of what if zPosition happens to be used in the future for the same reason, I would recommend using https://developer.apple.com/documentation/uikit/uiview/1622541-bringsubviewtofront